### PR TITLE
[PXN-2407] - Tracking adjustments

### PR DESCRIPTION
**Pull request by victorpereiradep**
## What have changed?
Changes were made to remove unnecessary data that could be passed in tracking and some improvements related to tracking.

## Kind of pr:

- [x] BugFix
- [ ] Feature
- [ ] Improvement

## How to test:
To test, just go through the streams and check if the data is being successfully validated based on melidata, this information stays in the Xcode log with the message `validated succesfully against melidata catalog`

## Extras
- IT NEEDS A ROBUST EVALUATION TO ENSURE THAT THE LOGICS WERE MAINTAINED AFTER THE CHANGES.
- `externalData` changes based on [PX - Melidata catalog](https://github.com/mercadolibre/fury_melidata-catalog-definitions/blob/master/src/main/resources/catalog/melidata/px.groovy)

### CODE REVIEW

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift [hunk 1]
```diff
@@ -79,67 +79,79 @@ internal extension MPXTracker {
...
+    }
+    
+    private func appendExternalData(to metadata: [String: Any]) -> [String: Any] {
+        var metadata = metadata
+        
+        if let flowDetails = flowDetails {
+            metadata["flow_detail"] = flowDetails
+        }
+        
+//            metadata["collector_id"] =
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift:98]: Remove this comment if we are not gonna use it

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift [hunk 2]
```diff
@@ -79,67 +79,79 @@ internal extension MPXTracker {
...
+        metadata[SessionService.SESSION_ID_KEY] = getSessionID()
+        metadata["session_time"] = PXTrackingStore.sharedInstance.getSecondsAfterInit()
+        
+        if let checkoutType = PXTrackingStore.sharedInstance.getChoType() {
+            metadata["checkout_type"] = checkoutType
+        }
+        
+        metadata["security_enabled"] = PXConfiguratorManager.hasSecurityValidation()
+        
+//            externalData["device_secured"] =
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift:109]: Remove this comment if we are not gonna use it

Comment (victorpereiradep) [reply MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift:109]: Nossa, PR é tudo, vc sabe onde eu consigo essa informação?

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/TrackingEvents.swift [hunk 3]
```diff
@@ -10,4 +10,5 @@ import Foundation
 protocol TrackingEvents {
     var name: String { get }
     var properties: [String : Any] { get }
+    var needsExternalData: Bool { get }
```
Comment (mltheus) [inline MercadoPagoSDK/MercadoPagoSDK/Tracking/TrackingEvents.swift:13]: can you explain your approach to add this bool without removing `properties`

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift (hunk 1 above)
Comment (mltheus) [reply MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift:98]: Mano, não sei =(

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift (hunk 2 above)
Comment (mltheus) [reply MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift:109]: não sei também =(

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/TrackingEvents.swift (hunk 3 above)
Comment (victorpereiradep) [reply MercadoPagoSDK/MercadoPagoSDK/Tracking/TrackingEvents.swift:13]: Some trackings needs `externalData`, so I control it with this parameter, you can check https://github.com/mercadolibre/fury_melidata-catalog-definitions/blob/master/src/main/resources/catalog/melidata/px.groovy to see this trackings

File: MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift (hunk 2 above)
Comment (victorpereiradep) [reply MercadoPagoSDK/MercadoPagoSDK/Tracking/MPXTracker.swift]: essa é só pra android o Cristion explicou

**Review by mltheus — APPROVED** [review]

**Pull request merged**